Deprecate extendPrimitive (#233)
radix-ui/website: data/primitives/overview/styling.mdx

## Extending a primitive

> Extending a primitive is done the same way you extend any React component. One minor difference is the use of `Radix.ComponentPropsWithoutRef` instead of `React.ComponentPropsWithoutRef` which is just a workaround until [the fixed types are released](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55396).

Reviewer: ```suggestion
Extending a primitive is done the same way you extend any React component. One minor difference is the use of `Radix.ComponentPropsWithoutRef` instead of `React.ComponentPropsWithoutRef` while we await review of a [`@types/react` update](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55396).
```

the main change I'm trying to propose is to remove  "the fixed types" part because there is a chance there is a reason the types are the way they are on DefinitelyTyped and my PR may get rejected.

It might not be likely but probably better off not making assumptions in our wording.

Reviewer: Oh, having said all that I've just seen that they've [gotten around to reviewing it now](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55396#issuecomment-911796016) and sounds like it might be a goer. So, feel free to ignore this!

Author: Yeah I checked before and it seems they agree on the correctness of your fix, so I thought I'd mention here.
